bridge_linux.go

```diff
@@ -105,7 +129,20 @@ func (h *Handle) bridgeVlanModify(cmd int, link Link, vid uint16, pvid, untagged
 	if untagged {
 		vlanInfo.Flags |= nl.BRIDGE_VLAN_INFO_UNTAGGED
 	}
-	br.AddRtAttr(nl.IFLA_BRIDGE_VLAN_INFO, vlanInfo.Serialize())
+
+	if vidEnd != 0 {
+		vlanEndInfo := &nl.BridgeVlanInfo{Vid: vidEnd}
+		vlanEndInfo.Flags = vlanInfo.Flags
+
+		vlanInfo.Flags |= nl.BRIDGE_VLAN_INFO_RANGE_BEGIN
+		br.AddRtAttr(nl.IFLA_BRIDGE_VLAN_INFO, vlanInfo.Serialize())
+
+		vlanEndInfo.Flags |= nl.BRIDGE_VLAN_INFO_RANGE_END
+		br.AddRtAttr(nl.IFLA_BRIDGE_VLAN_INFO, vlanEndInfo.Serialize())
+	} else { 
+		br.AddRtAttr(nl.IFLA_BRIDGE_VLAN_INFO, vlanInfo.Serialize())
```

Comment:
according to the testing.
We can not add "vlanInfo" after the "vlanEndInfo".
It will cause that only the last vlan will be added into the bridge.
e.g. add vlan 4~6, but only vlan6 is added.

Thank you!

Reply:
Thank you for checking.